registry/storage/filewriter_test.go

```diff
@@ -165,8 +165,9 @@ func TestSimpleWrite(t *testing.T) {
 
 func TestBufferedFileWriter(t *testing.T) {
 	ctx := context.Background()
-	writer, err := newFileWriter(ctx, inmemory.New(), "/random")
 
+	writerp, err := newFileWriter(ctx, inmemory.New(), "/random")
+	writer := *writerp
```

Comment:
So, you're saying that updates are not propagated here:
https://github.com/aibaars/distribution/blob/9f83bfc071094019b942c4332d17327b4dc5cdfc/registry/storage/blobwriter.go#L103.

> I opted for fixing the problem in bufferedFileWriter, because I think it is not nice to have a type that exhibits subtle misbehaviour when used by value instead of by reference.

I'm not sure if this is a valid conclusion. Updates aren't propagating to pointer mismanagement. I suspect a missing method somewhere, but its not clear. The copy of `bufferedFileWriter` should be fine when embedded.

It sounds like `(*blobWriter).Write` is broken because the multiwriter isn't taking the correct address. I haven't been able to confirm.